apps/E2E/src/common/BasePage.ts

```diff
@@ -209,33 +213,86 @@ export abstract class BasePage {
     });
   }
 
-  /* Scrolls to the specified or primary UI test element until it is displayed. */
-  async scrollToTestElement(component?: WebdriverIO.Element): Promise<void> {
-    const ComponentToScrollTo = component ?? (await this._primaryComponent);
-    if (await ComponentToScrollTo.isDisplayed()) {
+  /* Scrolls to the specified or primary UI test element until it is displayed.
+   * Either the component identifier or the component itself can be passed in as a parameter.
+   * On Android only the component identifier can be passed in.
+   * If no parameter is passed in, the primary component is used.
+   */
+  async scrollToTestElement(component?: WebdriverIO.Element | string): Promise<void> {
```

Comment:
I don't think we should allow the `| string` here. It adds complexity and confusion as to what a user _should_ use.

Is this specific to Android? If you need the testID, I believe you can reverse-engineer from the WebDriverIO.Element to get the automationID

Reply:
The string is required for Android.
It won't be possible to get the ID from the component because the component itself is not retrievable when using By() when it is not on the screen!

Reply:
I could fork the function just for Android.
Alternatively, I can change the function to only take a string, the webdriverio.element will be accessible from it, and we do have string IDs available everywhere. 
Which approach do you suggest?

Reply:
Let's create an Overload of the function. One takes WebDriverIO.Element and one takes a string

Reply:
And let's make sure the comments explain why we have two different functions, and which platforms should use which

Reply:
Moving ahead as discussed offline with separate mobile scroll function since the parameter types are different and have limited code duplication. Also simplifies the functions and will be easier to diagnose misused function platform wise while writing tests.